### PR TITLE
Added xml_uri generation

### PR DESCRIPTION
**Pull request by seanwiseman**
A task from https://github.com/elifesciences/api-raml/pull/215.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -169,6 +169,14 @@ def pdf_uri(triple):
     filename = "elife-%s-v%s.pdf" % (utils.pad_msid(msid), version) # ll: elife-09560-v1.pdf
     return cdnlink(msid, filename)
 
+def xml_uri(triple):
+    """predict an article's xml url."""
+    content_type, msid, version = triple
+    if content_type and any(lmap(lambda type: type in ['Correction', 'Retraction'], content_type)):
```
Comment (giorgiosironi) [inline src/main.py]: this is true for PDF (where this comes from) but I think the XML is always there for any article. I'll check.

Comment (seanwiseman) [reply src/main.py]: Ok thank you.

File: src/main.py [hunk 2]
```diff
@@ -169,6 +169,14 @@ def pdf_uri(triple):
...
+        return EXCLUDE_ME
+    filename = "elife-%s-v%s.xml" % (utils.pad_msid(msid), version) # ll: elife-09560-v1.xml
```
Comment (giorgiosironi) [inline src/main.py]: it is correct to pad this:
```
[14:34:25][giorgio@Newton:~/code/builder]$ aws s3 ls prod-elife-published/articles/09560/ | grep xml
2017-03-29 06:47:41     249611 elife-09560-v1-download.xml
2017-03-29 06:47:41     249611 elife-09560-v1.xml
```
:+1:

File: src/main.py [hunk 3]
```diff
@@ -169,6 +169,12 @@ def pdf_uri(triple):
     filename = "elife-%s-v%s.pdf" % (utils.pad_msid(msid), version) # ll: elife-09560-v1.pdf
     return cdnlink(msid, filename)
 
+def xml_uri(triple):
+    """predict an article's xml url."""
+    content_type, msid, version = triple
```
Comment (giorgiosironi) [inline src/main.py]: content-type is not needed anymore here

Comment (seanwiseman) [reply src/main.py]: You are right. I have now removed it. I felt pressured to keep it due to the name `triple` 😱.

**Pull request merged**